### PR TITLE
Remove autotools files from gitignore

### PR DESCRIPTION
**Pull request by yol**
Remove autotools files from gitignore

## Motivation and Context
Files from autotools are still in gitignore. They shouldn't be there so people can clean up old cruft with `git clean`.

## Types of change

- [ ] Bug fix (non-breaking change which fixes an issue)
- [ ] New feature (non-breaking change which adds functionality)
- [ ] Breaking change (fix or feature that would cause existing functionality to change)

## Checklist:

- [x] My code follows the [Code guidelines](https://codedocs.xyz/xbmc/xbmc/code_guidelines.html) of this project 
- [ ] My change requires a change to the documentation, either Doxygen or wiki
- [ ] I have updated the documentation accordingly
- [x] I have read the [CONTRIBUTING](https://github.com/xbmc/xbmc/blob/master/CONTRIBUTING.md) document
- [ ] I have added tests to cover my change
- [ ] All new and existing tests passed

### CODE REVIEW

**Comment by pkerling** [comment]
Maybe there's more that can be removed, input appreciated.

File: .gitignore [hunk 1]
```diff
@@ -18,9 +18,6 @@ testMain
 .vscode/
 
 # generated files etc
-config.cache
-config.status
-config.log
```
Comment (wsnipex) [inline .gitignore]: those 3 still exist in tools/depends

Comment (pkerling) [inline .gitignore]: So would adding `/tools/depends/**/config.cache` etc. be OK?

Comment (wsnipex) [inline .gitignore]: sure

Comment (pkerling) [inline .gitignore]: done

File: .gitignore [hunk 2]
```diff
@@ -133,9 +109,26 @@ cmake_install.cmake
...
+/addons/kodi.binary.instance.audioencoder/addon.xml
+/addons/kodi.binary.instance.game/addon.xml
+/addons/kodi.binary.instance.imagedecoder/addon.xml
+/addons/kodi.binary.instance.inputstream/addon.xml
+/addons/kodi.binary.instance.peripheral/addon.xml
+/addons/kodi.binary.instance.pvr/addon.xml
+/addons/kodi.binary.instance.screensaver/addon.xml
+/addons/kodi.binary.instance.vfs/addon.xml
+/addons/kodi.binary.instance.videocodec/addon.xml
+/addons/kodi.binary.instance.visualization/addon.xml
```
Comment (wsnipex) [inline .gitignore]: does it make sense to use a wildcard here?
i.e. /addons/kodi.binary.instance.*/addon.xml

Comment (pkerling) [inline .gitignore]: Why are those needed at all? Aren't they supposed to go to the build folder anyway?

Comment (pkerling) [inline .gitignore]: friendly ping :-)

Comment (Rechi) [inline .gitignore]: that's for people building in-source
as we warn about such builds at https://github.com/xbmc/xbmc/blob/d790e54/CMakeLists.txt#L14-L16 imo you can remove it completely from .gitignore

@garbear do you build in-source?

Comment (garbear) [inline .gitignore]: On windows I build within a build/ subfolder, but these are generated anyway

Comment (Rechi) [inline .gitignore]: If you don't build in-source, they are generated in your build subfolder which is already completely ignored.

Comment (garbear) [inline .gitignore]: Oh right, I symlink /build/addons to /addons to avoid having to recompile binary add-ons (a big pain with 60 game add-ons) whenever I remove and recreate my build folder

Comment (notspiff) [inline .gitignore]: would you accept a fix for the addon macros to do the same? I hate the write to source tree there just as much..

Comment (hudokkow) [inline .gitignore]: Yes please! 😀

**Comment by AlwinEsch** [comment]
From my side OK

**Comment by Razzeee** [comment]
@pkerling 
Do you mind if we squash this down to one commit on merge?

**Comment by pkerling** [comment]
@Razzeee no, I think that's a good idea
